leetcode/227_basic_calculator_ii.py

```diff
@@ -0,0 +1,83 @@
+class Solution:
+
+    def calculate(self, s: str) -> int:
+        """Evaluates a mathematical expression given as a string using a stack.
+
+        Args:
+            s (str): A string containing the mathematical expression.
+
+        Returns:
+            int: The result of evaluating the expression.
+        """
+
+        digits = {'0':0, '1':1, '2':2, '3':3, '4':4, '5':5, '6':6, '7':7, '8':8, '9':9}
```

Comment:
I like this. this looks much cleaner compared to casting it every time with `int()`. nice!